core/server/src/main/java/alluxio/worker/netty/DataServerFileReadHandler.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.worker.netty;
+
+import alluxio.Constants;
+import alluxio.metrics.MetricsSystem;
+import alluxio.network.protocol.RPCProtoMessage;
+import alluxio.network.protocol.databuffer.DataBuffer;
+import alluxio.network.protocol.databuffer.DataNettyBufferV2;
+import alluxio.proto.dataserver.Protocol;
+import alluxio.worker.file.FileSystemWorker;
+
+import com.codahale.metrics.Counter;
+import io.netty.buffer.ByteBuf;
+import io.netty.channel.Channel;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.concurrent.ExecutorService;
+
+import javax.annotation.concurrent.NotThreadSafe;
+
+/**
+ * This handler handles file read request. Check more information in {@link DataServerReadHandler}.
+ */
+@NotThreadSafe
+public final class DataServerFileReadHandler extends DataServerReadHandler {
```

Comment:
Consider `DataServerUfsFileReadHandler` or `DataServerUfsReadHandler`
same for the write handler